conformance/tests/httproute-request-percentage-mirror.go

```diff
@@ -0,0 +1,236 @@
+							defer wg.Done()
+							defer func() { <-semaphore }()
+							http.MakeRequestAndExpectEventuallyConsistentResponse(t, r, timeoutConfig, gwAddr, expected)
+						}(t, suite.RoundTripper, suite.TimeoutConfig, gwAddr, expected)
+					}
+					wg.Wait()
+
+					if err := testMirroredRequestsDistribution(t, suite, expected, time); err != nil {
+						t.Logf("Traffic distribution test failed (%d/%d): %s", k+1, numDistributionChecks, err)
+					} else {
+						return
+					}
+				}
+				t.Fatal("Percentage based mirror distribution tests failed")
+			})
+		}
+	},
+}
+
+func testMirroredRequestsDistribution(t *testing.T, suite *suite.ConformanceTestSuite, expected http.ExpectedResponse, timeVal time.Time) error {
+	mirrorPods := expected.MirroredTo
+	for i, mirrorPod := range mirrorPods {
+		if mirrorPod.Name == "" {
+			tlog.Fatalf(t, "Mirrored BackendRef[%d].Name wasn't provided in the testcase, this test should only check http request mirror.", i)
+		}
+	}
+
+	var mu sync.Mutex
+	mirroredCounts := make(map[string]int)
+
+	for _, mirrorPod := range mirrorPods {
+		require.Eventually(t, func() bool {
+			mirrorLogRegexp := regexp.MustCompile(fmt.Sprintf("Echoing back request made to \\%s to client", expected.Request.Path))
+
+			tlog.Log(t, "Searching for the mirrored request log")
+			tlog.Logf(t, `Reading "%s/%s" logs`, mirrorPod.Namespace, mirrorPod.Name)
+			logs, err := kubernetes.DumpEchoLogs(mirrorPod.Namespace, mirrorPod.Name, suite.Client, suite.Clientset, timeVal)
+			if err != nil {
+				tlog.Logf(t, `Couldn't read "%s/%s" logs: %v`, mirrorPod.Namespace, mirrorPod.Name, err)
+				return false
+			}
+
+			count := 0
+			for _, log := range logs {
+				if mirrorLogRegexp.MatchString(log) {
+					count++
+				}
+			}
+			mu.Lock()
+			mirroredCounts[mirrorPod.Name] += count
+			mu.Unlock()
+
+			return true
+		}, 60*time.Second, time.Millisecond*100, fmt.Sprintf(`Couldn't verify the logs for "%s/%s"`, mirrorPod.Namespace, mirrorPod.Name))
+	}
+
+	var errs []error
+
+	for _, mirrorPod := range mirrorPods {
+		expected := float64(totalRequests) * float64(*mirrorPod.Percent) / 100.0
+		minExpected := expected * (1 - tolerancePercentage/100)
+		maxExpected := expected * (1 + tolerancePercentage/100)
+
+		actual := float64(mirroredCounts[mirrorPod.Name])
+		tlog.Logf(t, "Pod: %s, Expected: %f (min: %f, max: %f), Actual: %f", mirrorPod.Name, expected, minExpected, maxExpected, actual)
+
+		if actual < minExpected || actual > maxExpected {
+			errs = append(errs, fmt.Errorf("Pod %s did not meet the mirroring percentage within tolerance. Expected between %f and %f, but got %f", mirrorPod.Name, minExpected, maxExpected, actual))
+		}
+	}
+	if len(errs) > 0 {
+		return errors.Join(errs...)
+	}
+	tlog.Log(t, "Validated mirrored request logs across all desired backends within the given tolerance")
+	return nil
+}
+
+func ptrTo[T any](a T) *T {
+	return &a
+}
```

Comment:
You can use the package `k8s.io/utils/ptr` instead of defining your own function